src/unpack/unpack_0_127.ts

```diff
@@ -64,17 +63,20 @@ export default function (incoming: Buffer, isoJSON: Types.KeyValueStringT, confi
             if (thisLen === 0) {
               return T.toErrorObject(['field ', field, ' format not implemented']);
             } else {
-              const len = thisBuff.slice(0, thisLen).toString();
-              thisBuff = thisBuff.slice(thisLen, thisBuff.byteLength);
-              isoJSON[field] = thisBuff.slice(0, Number(len)).toString();
-              thisBuff = thisBuff.slice(Number(len), thisBuff.byteLength);
+              if (config.custom127Encoding && field === 127) {
+                isoJSON[field] = thisBuff.slice(0).toString(config.custom127Encoding);
+              } else {
+                const len = thisBuff.slice(0, thisLen).toString();
+                thisBuff = thisBuff.slice(thisLen, thisBuff.byteLength);
+                isoJSON[field] = thisBuff.slice(0, Number(len)).toString(config.bitmapEncoding || 'hex');
+                thisBuff = thisBuff.slice(Number(len), thisBuff.byteLength);
```

Comment:
should be a sub function returnigin an arra [value, tailBuff ]
would probably make it easier to read and you could write some meaningful tests for that.